src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -79,62 +100,131 @@ public T runWithAnyNode() {
     }
   }
 
-  private T runWithRetries(final int slot, int attempts, boolean tryRandomNode, JedisRedirectionException redirect) {
-    if (attempts <= 0) {
-      throw new JedisClusterMaxAttemptsException("No more cluster attempts left.");
+  private boolean shouldBackOff(int attemptsLeft) {
+    int attemptsDone = maxAttempts - attemptsLeft;
+    return attemptsDone >= maxAttempts / 3;
+  }
```

Comment:
I don't agree with the math here. It'd be a long comment to explain. Rather, let me find some time and put my spin over this PR.

Reply:
I invited you as a collaborator to https://github.com/walles/jedis so that you can push to the PR branch.